stock_picking_purchase_order_alert/models/stock_picking.py

```diff
@@ -0,0 +1,153 @@
+        logger.info(groups_names)
+        logger.info(",".join(groups_names))
+        return ",".join(groups_names)
+
+    def _get_quantity_alert_lines(self, picking):
+        """Get lines that exceed the configured threshold percentage."""
+        alert_lines = []
+        alert_pct = picking.picking_type_id.quantity_alert_percentage
+        threshold_factor = 1 + (alert_pct / 100.0)
+
+        moves_with_qty = picking.move_ids_without_package.filtered(
+            lambda m: m.purchase_line_id and m.quantity_done > 0
+        )
+
+        for move in moves_with_qty:
+            po_line = move.purchase_line_id
+            po_qty = po_line.product_qty
+
+            if not po_qty > 0:
+                continue
+
+            received_qty = move.product_uom._compute_quantity(
+                move.quantity_done, po_line.product_uom
+            )
+
+            if received_qty <= po_qty * threshold_factor:
+                continue
+
+            excess_percent = (received_qty / po_qty - 1) * 100
+            alert_lines.append(
+                {
+                    "product": move.product_id.display_name,
+                    "ordered": po_qty,
+                    "ordered_uom": po_line.product_uom.name,
+                    "received": received_qty,
+                    "received_uom": po_line.product_uom.name,
+                    "percent": excess_percent,
+                    "threshold": picking.picking_type_id.quantity_alert_percentage,
+                }
+            )
+
+        return alert_lines
+
+    def _format_quantity_alert_message(self, alert_percentage, alert_lines):
+        """Format the HTML message for quantity alerts."""
+        message = (
+            "<div class='alert alert-warning'>"
+            "<strong>Warning!</strong> Material with quantity "
+            "exceeding ordered amount detected (threshold: "
+            f"{alert_percentage}%): <br/>"
+            "<ul>"
+        )
+
+        msg_line_template = (
+            "<li><strong>{product}</strong>: "
+            "Ordered: {ordered} {ordered_uom}, "
+            "To receive: {received} {received_uom} "
+            "(excess of {percent:.2f}%)</li>"
+        )
+
+        message += "".join(msg_line_template.format(**line) for line in alert_lines)
+        message += "</ul></div>"
+
+        return message
+
+    def button_validate(self):
+        """Override button_validate to prevent validation
+        when quantity alerts exist"""
+        for picking in self:
+            groups_ids = picking.picking_type_id.groups_ids
+            if groups_ids and not (
+                self.env.user.user_has_groups(self._get_group_names(groups_ids.ids))
+            ):
+                raise UserError(
+                    _(
+                        f"Only the users that belongs to this groups {','.join(groups_ids.mapped('name'))} can validate the picking"
+                    )
+                )
```

Comment:
This way it will be printed like:
`In order to validate the picking you need to belong to any of the groups:`
`- Purchase / Administrator`
```suggestion
                group_names = [f"- {group.display_name}" for group in groups_ids]
                error_msg = _(
                    "In order to validate the picking you need to belong to any of the groups:\n%s"
                ) % "\n".join(group_names)
                raise UserError(error_msg)
```